### PR TITLE
Add mrqa data loader

### LINKED ISSUE #32: Add Dataloader - TF Dataset
Add a method for loading the data into a tf.Dataset.

### PR DESCRIPTION
**Pull request by afansi**
This PR adds a dataloader to the project. To achieve it, we did the following steps:

1- Add utility methods for textual data pre-processing
2- Add a structure that represents a MRQA example
3- Add a class for reading MRQA Datasets
4- Add a class for representing Input Feature data
5- Add a method for converting MRQA Example  into Input Feature data
6- Add a class for writing a collection of Input Feature data into a TFRecord data file
7- Add a method for loading TFRecord datafile and use it as a tf.Dataset
8- Add a function to manipulate the tf.Dataset with Repeat and Batching functionalities

Closes #32

### CODE REVIEW

File: unit_tests/test_data_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+def test_remove_html_tags():
+    '''Unit test of data utility'''
+    from jag.utils.data_utils import remove_html_tags
```
Comment (mirkobronzi) [inline unit_tests/test_data_utils.py]: any reason to have the import here instead of at the top of the file?

File: unit_tests/test_dataloader.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+def test_dataloader():
+    import tensorflow as tf
```
Comment (mirkobronzi) [inline unit_tests/test_dataloader.py]: same (why import not at the top?)

File: jag/data/data_handler.py [hunk 3]
```diff
@@ -0,0 +1,266 @@
+import gzip
```
[the remark this answers is not in the document]
Comment (afansi) [reply jag/data/data_handler.py]: Indeed, we got inspiration from the google Bert repository (https://github.com/google-research/bert/blob/master/run_squad.py) and the code were adapted to read the MRQA Dataset.
I added this reference as a comment in the code as you suggested.

File: jag/data/data_handler.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+                    context_text=raw_context,
+                    context_tokens=context_tokens,
+                    answers_text=effective_answer_list,  # answer_list,
+                    contextualized_answers=answers,
+                    start_position=start_position,
+                    end_position=end_position,
+                )
+                examples.append(example)
+
+        print('escape examples: ', escape_examples)
```
[the remark this answers is not in the document]
Comment (afansi) [reply jag/data/data_handler.py]: I replace the "print" method with this logging features. Thanks for the advice. :)

File: unit_tests/test_data_utils.py (hunk 1 above)
Comment (afansi) [reply unit_tests/test_data_utils.py]: I moved the import at the top of the files.

File: unit_tests/test_dataloader.py (hunk 2 above)
Comment (afansi) [reply unit_tests/test_dataloader.py]: Already addressed!

**Pull request merged**